### PR TITLE
Adds New Changeling Ability: Miasmic Mist

### PR DESCRIPTION
**Pull request by ItsMarmite**
## What Does This PR Do
This PR adds a new defensive changeling ability, Miasmic Mist. Costing 2 DNA to purchase, and taking 15 chemicals per use.

Miasmic Mist will produce obfuscating purple smoke that will slur the speech, and blur the vision of those who do not wear gas masks while in it. Additionally it will cause minor toxin and cause them to drop items from their hands.

Miasmic Mist immunity can also be gotten from purchasing the ability. This does mean the ability affects changelings who have not purchased Miasmic Mist.

## Why It's Good For The Game
Changeling heavily lacks a lot of abilities for disengaging and dealing with large groups. My hope with this ability is it will allow changelings to cover their tracks more, and allow them more leeway in avoiding 9 man security baton beatdowns by hiding in THE FOG. Additionally providing synergy with augmented eyesight, which allows a changeling to see while within THE FOG.

## Testing
https://github.com/ParadiseSS13/Paradise/assets/85680653/4225a183-1ba4-4d94-a6ec-387767193b7c

## Changelog
:cl:
add: Adds new defensive Changeling ability, Miasmic Mist. An AOE poison cloud that blinds its victims.
/:cl:

### CODE REVIEW

**Comment by spacegeese** [comment]
is it a bad idea to make the mist extinguish fires? seems fitting for an antag whose source material hates fire, plus it doesn't really change all that much if you have a pocket extinguisher

**Comment by Coolrune206** [comment]
Standard sec equipment includes sec gas mask. Does this ability bypass that or does the sec mask render it moot.

**Comment by ItsMarmite** [comment]
> Standard sec equipment includes sec gas mask. Does this ability bypass that or does the sec mask render it moot.

Yes, Security's sechailer masks will remain a counter to the item dropping/poison/etc side of the cloud although thermals would be required to see the changeling through the cloud.

As with most things, Security are uniquely more equipped to handle changelings in ways that, say, the average medical doctor may not be.

File: code/game/objects/effects/effect_system/effects_smoke.dm [hunk 1]
```diff
@@ -206,3 +206,38 @@
...
+		M.Jitter(10 SECONDS)
+		M.AdjustEyeBlurry(10 SECONDS)
+		M.AdjustDizzy(5 SECONDS)
+		M.AdjustSlur(10 SECONDS)
+		M.emote(pick("drool","quiver","tremble","scream"))
+		M.fakevomit()
+		M.adjustToxLoss(2)
+		M.drop_l_hand()
+		M.drop_r_hand()
+		return 1
```
Comment (Burzah) [inline code/game/objects/effects/effect_system/effects_smoke.dm:233]:
```suggestion
		return TRUE
```

File: code/game/objects/effects/effect_system/effects_smoke.dm [hunk 2]
```diff
@@ -206,3 +206,38 @@
...
+
+/obj/effect/particle_effect/smoke/changeling/CanPass(atom/movable/mover, turf/target, height=0)
+	if(height==0)
+		return 1
```
Comment (Burzah) [inline code/game/objects/effects/effect_system/effects_smoke.dm:237]:
```suggestion
	if(height == 0)
		return TRUE
```

File: code/game/objects/effects/effect_system/effects_smoke.dm [hunk 3]
```diff
@@ -206,3 +206,38 @@
...
+	if(istype(mover, /obj/item/projectile/beam))
+		var/obj/item/projectile/beam/B = mover
+		B.damage = (B.damage/2)
+	return 1
```
Comment (Burzah) [inline code/game/objects/effects/effect_system/effects_smoke.dm:241]:
```suggestion
		B.damage = (B.damage / 2)
	return TRUE
```

**Comment by Daylight2** [comment]
After testing it, this feels very strong in combat. It costs 6 ability points, sure, but the smoke goes over everyone's sprites so that you have to spamclick blindly and pray you hit the cling instead of another officer, whilst the cling can see just fine. And god forbid that you don't like the drip of the black gasmask and aren't wearing it at the time.

File: code/game/objects/effects/effect_system/effects_smoke.dm [hunk 4]
```diff
@@ -206,3 +206,38 @@
...
+/////////////////////////////////////////////
+// Changeling smoke
+/////////////////////////////////////////////
+
+/obj/effect/particle_effect/smoke/changeling
+	color = "#7A61B0"
+	lifetime = 10
+
+/obj/effect/particle_effect/smoke/changeling/smoke_mob(mob/living/carbon/M)
+	if(..())
```
Comment (Sirryan2002) [inline code/game/objects/effects/effect_system/effects_smoke.dm:219]: make this into a guard statement instead

File: code/game/objects/effects/effect_system/effects_smoke.dm [hunk 5]
```diff
@@ -206,3 +206,38 @@
...
+		M.fakevomit()
+		M.adjustToxLoss(2)
+		M.drop_l_hand()
+		M.drop_r_hand()
+		return TRUE
+
+/obj/effect/particle_effect/smoke/changeling/CanPass(atom/movable/mover, turf/target, height=0)
+	if(height == 0)
+		return TRUE
+	if(istype(mover, /obj/item/projectile/beam))
```
Comment (Sirryan2002) [inline code/game/objects/effects/effect_system/effects_smoke.dm:238]: is there a istype DEFINE for beams?

File: code/game/gamemodes/changeling/changeling.dm [hunk 6]
```diff
@@ -104,3 +104,9 @@ GLOBAL_LIST_INIT(possible_changeling_IDs, list("Alpha","Beta","Gamma","Delta","E
 				text += "<br><font color='red'><B>The changeling has failed.</B></font>"
 				SSblackbox.record_feedback("tally", "changeling_success", 1, "FAIL")
 		return text.Join("")
+
+/datum/antagonist/changeling/proc/get_ability(path)
+	for(var/datum/action/changeling/power in acquired_powers)
+		if(power.type == path)
+			return power
+	return null
```
Comment (Sirryan2002) [inline code/game/gamemodes/changeling/changeling.dm:112]:
```suggestion
```
hanging return

File: code/game/objects/effects/effect_system/effects_smoke.dm [hunk 7]
```diff
@@ -206,3 +206,38 @@
...
+		M.AdjustDizzy(5 SECONDS)
+		M.AdjustSlur(10 SECONDS)
+		M.emote(pick("drool","quiver","tremble","scream"))
+		M.fakevomit()
+		M.adjustToxLoss(2)
+		M.drop_l_hand()
+		M.drop_r_hand()
+		return TRUE
+
+/obj/effect/particle_effect/smoke/changeling/CanPass(atom/movable/mover, turf/target, height=0)
```
Comment (Sirryan2002) [inline code/game/objects/effects/effect_system/effects_smoke.dm:235]:
```suggestion
/obj/effect/particle_effect/smoke/changeling/CanPass(atom/movable/mover, turf/target, height = 0)
```

File: code/game/objects/effects/effect_system/effects_smoke.dm [hunk 8]
```diff
@@ -206,3 +206,38 @@
...
+	if(height == 0)
+		return TRUE
+	if(istype(mover, /obj/item/projectile/beam))
+		var/obj/item/projectile/beam/B = mover
+		B.damage = (B.damage / 2)
+	return TRUE
+/datum/effect_system/smoke_spread/changeling
```
Comment (Sirryan2002) [inline code/game/objects/effects/effect_system/effects_smoke.dm:242]:
```suggestion
	return TRUE
	
/datum/effect_system/smoke_spread/changeling
```
spacing

File: code/game/objects/effects/effect_system/effects_smoke.dm (hunk 4 above)
Comment (Sirryan2002) [reply code/game/objects/effects/effect_system/effects_smoke.dm:219]: still would like to see this implemented

**Review by Sirryan2002 — CHANGES_REQUESTED** [review]

File: code/game/objects/effects/effect_system/effects_smoke.dm (hunk 4 above)
Comment (ItsMarmite) [reply code/game/objects/effects/effect_system/effects_smoke.dm]: Made some changes.

File: code/game/objects/effects/effect_system/effects_smoke.dm [hunk 9]
```diff
@@ -218,3 +218,42 @@
...
+/////////////////////////////////////////////
+// Changeling smoke
+/////////////////////////////////////////////
+
+/obj/effect/particle_effect/smoke/changeling
+	color = "#7A61B0"
+	lifetime = 10
+
+/obj/effect/particle_effect/smoke/changeling/smoke_mob(mob/living/carbon/M)
+	if (!..())
```
Comment (Burzah) [inline code/game/objects/effects/effect_system/effects_smoke.dm:231]:
```suggestion
	if(!..())
```

File: code/game/objects/effects/effect_system/effects_smoke.dm [hunk 10]
```diff
@@ -218,3 +218,42 @@
...
+		return
+
+	var/datum/antagonist/changeling/C = M.mind?.has_antag_datum(/datum/antagonist/changeling)
+	if (C && C.get_ability(/datum/action/changeling/miasmic_mist))
```
Comment (Burzah) [inline code/game/objects/effects/effect_system/effects_smoke.dm:235]:
```suggestion
	if(C && C.get_ability(/datum/action/changeling/miasmic_mist))
```

File: code/modules/antagonists/changeling/powers/miasmic_mist.dm [hunk 11]
```diff
@@ -0,0 +1,19 @@
...
+	dna_cost = 2
+	req_human = TRUE
+	power_type = CHANGELING_PURCHASABLE_POWER
+	category = /datum/changeling_power_category/defence
+
+/datum/action/changeling/miasmic_mist/sting_action(mob/living/carbon/user)
+	var/datum/effect_system/smoke_spread/changeling/smoke = new
+	smoke.set_up(10, FALSE, get_turf(user))
+	smoke.start()
+	sleep(10)
```
Comment (DGamerL) [inline code/modules/antagonists/changeling/powers/miasmic_mist.dm:16]: Could you turn this sleep into an `addtimer`?

**Review by Sirryan2002 — APPROVED** [review]
just see to DGamer's review

File: code/game/objects/effects/effect_system/effects_smoke.dm [hunk 12]
```diff
@@ -218,3 +218,42 @@
...
+	M.AdjustSlur(10 SECONDS)
+	M.emote(pick("drool", "quiver", "tremble", "scream"))
+	M.fakevomit()
+	M.adjustToxLoss(2)
+	M.drop_l_hand()
+	M.drop_r_hand()
+	return TRUE
+
+/obj/effect/particle_effect/smoke/changeling/CanPass(atom/movable/mover, turf/target, height = 0)
+	if(height == 0)
```
[the remark this answers is not in the document]
Comment (Contrabang) [reply code/game/objects/effects/effect_system/effects_smoke.dm:251]: surprisingly, not a true/false thing. This should be kept as `== 0`

**Review by DGamerL — CHANGES_REQUESTED** [review]
Please resolve the conflict and look at  the reviews, PR is getting quite stale

**Comment by SynthTwo** [comment]
!wiki_label

File: code/game/objects/effects/effect_system/effects_smoke.dm [hunk 13]
```diff
@@ -218,3 +218,42 @@
...
+
+	M.drop_item()
+	M.Jitter(10 SECONDS)
+	M.AdjustEyeBlurry(10 SECONDS)
+	M.AdjustDizzy(5 SECONDS)
+	M.AdjustSlur(10 SECONDS)
+	M.emote(pick("drool", "quiver", "tremble", "scream"))
+	M.fakevomit()
+	M.adjustToxLoss(2)
+	M.drop_l_hand()
```
Comment (lewcc) [inline code/game/objects/effects/effect_system/effects_smoke.dm:246]: Why drop_item(), drop_l_hand(), and drop_r_hand? Doesn't the cough emote auto stop stuff too?

File: code/game/objects/effects/effect_system/effects_smoke.dm [hunk 14]
```diff
@@ -218,3 +218,42 @@
...
+	M.drop_r_hand()
+	return TRUE
+
+/obj/effect/particle_effect/smoke/changeling/CanPass(atom/movable/mover, turf/target, height = 0)
+	if(height == 0)
+		return TRUE
+	if(istype(mover, /obj/item/projectile/beam))
+		var/obj/item/projectile/beam/B = mover
+		B.damage = (B.damage / 2)
```
Comment (lewcc) [inline code/game/objects/effects/effect_system/effects_smoke.dm:255]:
```suggestion
		B.damage /= 2
```

**Comment by DGamerL** [comment]
The PR is stale and the author has said that they do not have motivation to currently work on it

**Pull request closed without merging**